pallets/dapp-staking-v3/src/types.rs

```diff
@@ -589,6 +586,116 @@ where
 
         amount
     }
+
+    /// Active staked balance.
+    ///
+    /// In case latest stored information is from the past period, active stake is considered to be zero.
+    pub fn active_stake(&self, active_period: PeriodNumber) -> Balance {
+        match self.staked_period {
+            Some(last_staked_period) if last_staked_period == active_period => self
+                .staked
+                .0
+                .last()
+                .map_or(Balance::zero(), |chunk| chunk.amount),
+            _ => Balance::zero(),
+        }
+    }
+
+    /// Amount that is available for staking.
+    ///
+    /// This is equal to the total active locked amount, minus the staked amount already active.
+    pub fn stakeable_amount(&self, active_period: PeriodNumber) -> Balance {
+        self.active_locked_amount()
+            .saturating_sub(self.active_stake(active_period))
+    }
+
+    /// Amount that is staked, in respect to currently active period.
+    pub fn staked_amount(&self, active_period: PeriodNumber) -> Balance {
+        match self.staked_period {
+            Some(last_staked_period) if last_staked_period == active_period => self
+                .staked
+                .0
+                .last()
+                // We should never fallback to the default value since that would mean ledger is in invalid state.
+                // TODO: perhaps this can be implemented in a better way to have some error handling? Returning 0 might not be the most secure way to handle it.
```

Comment:
How about if we simply return `Option<Balance>` and let consumer (like `ContractStakingInfoSeries`) handle the None, like an error depending on the use?